src/components/Deployment/Deployment.tsx

```diff
@@ -0,0 +1,229 @@
+  const [undeployedChains, setUndeployedChains] = useState<DeployChain[]>([]);
+  const [confirmModal, setConfirmModal] = useState<boolean>(false);
+  const [selectedChain, setSelectedChain] = useState<number | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const { getSdkForChainId, accountAddress } = useEtherspot();
+  const theme: Theme = useTheme();
+
+  const fetchDeploymentData = async (chainId: number) => {
+    const sdk = getSdkForChainId(chainId);
+    if (accountAddress && sdk) {
+      try {
+        await sdk.computeContractAccount();
+        return await sdk.getAccount();
+      } catch (err) {
+        //
+      }
+    }
+  };
+
+  const getDeploymentStatus = async () => {
+    try {
+      const result = [];
+      for (const { chainId, title, iconUrl } of supportedChains) {
+        const account = await fetchDeploymentData(chainId);
+        if (account) {
+          result.push({ ...account, chainId, title, iconUrl });
+        }
+      }
+      return result;
+    } catch (err) {
+      return [];
+    }
+  };
+
+  const hideModal = () => {
+    setConfirmModal(false);
+  };
+
+  const deploymentData = async () => {
+    const data: DeployChain[] = await getDeploymentStatus();
+    if (!data) return;
+    const [unDeployed, deployed] = data.reduce(
+      (acc: DeployChain[][], curr: DeployChain) => {
+        if (curr.state ===  AccountStates.UnDeployed) {
+          acc[0].push(curr);
+        } else {
+          acc[1].push(curr);
+        }
+        return acc;
+      },
+      [[], []]
+    );
+    setUndeployedChains(unDeployed);
+    setDeployedChains(deployed);
+  };
+
+  const deploy = async () => {
+    if (!selectedChain || !accountAddress) return;
+    const sdk = getSdkForChainId(selectedChain);
+    if (!sdk) {
+      setErrorMessage('Failed to proceed with selected actions!');
+      return;
+    }
+    try {
+      await deployAccount(sdk);
+    } catch (err) {
+      setErrorMessage('Failed to proceed with selected actions!');
+      hideModal();
+    }
+  };
+
+  const handleClose = () => {
+    setErrorMessage('');
+  };
+
+  useEffect(() => {
+    deploymentData();
+  }, []);
+
+  return (
+    <Card title="Deployments" marginBottom={20}>
+      {errorMessage && <ErrorMessage errorMessage={errorMessage} onClose={handleClose} />}
+      {!!confirmModal && (
+        <Modal isComponentOnTop>
+          <Paragraph>Deploy your wallet on Ethereum to sign messages</Paragraph>
+          <PrimaryButton
+            background={theme?.color?.background?.selectInputScrollbar}
+            color={theme?.color?.text?.listItemQuickButtonPrimary}
+            display="flex"
+            justifyContent="center"
+            onClick={deploy}
+          >
+            Deploy
+          </PrimaryButton>
+          <br />
+          <SecondaryButton margin="0 auto" display="flex" color={theme.color?.text?.card} onClick={hideModal}>
```

Comment:
if this is component specific we might Just create a Wrapper Styled component that Styles the `SecondaryButton` component instead of passing all these properties. Just a suggestion